### ISSUE TITLE
numpy has teriminated the `get_info` method

### ISSUE THREAD

**Issue by ankarsingh309600** [benfred/implicit#691]
In the latest versions of `numpy>1.24.0`, `np.__config__.get_info` method has been terminated, which is causing the follwoing error when running the `ALS`:

`AttributeError: module 'numpy.__config__' has no attribute 'get_info'`.
The functionality is used to detect `OpenBLAS` and `Intel MKL BLAS`.

Simple fix is to use the function `get_info` which numpy has deprecated.

```python
def get_info(name):
    """Check given global attribute."""
    g = globals()
    return g.get(name, g.get(name + "_info", {}))
```

**Comment by nwittenberg** [comment]
Upvote. Facing same issue - now we have to manage numpy dependency explicitly. Would be great if this could be addressed.

**Comment by benfred** [comment]
I think we should probably remove this block of code. We're trying to detect a potential performance problem (having BLAS configured to use a threadpool causes severe perf isues since we are also internally multi-threading which leads to thrashing- so we want to make sure that BLAS is configured to just use one thread, and we're using this method to try to figure out which BLAS library is being used by numpy).  

Instead I think we should use https://github.com/joblib/threadpoolctl here - which both lets us get the threadpool status for the BLAS libraries, as well as programmatically set it. This means rather than just detecting the perf issue, we can configure BLAS appropriately to not have the issue in the first place.

**Comment by benfred** [comment]
https://github.com/benfred/implicit/pull/692

**Comment by benfred** [comment]
Fix will be in the 0.7.2 release - which will be pushed out by CI in the next hour or so
